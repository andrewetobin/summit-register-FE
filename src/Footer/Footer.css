.Footer {
  background: rgba(165, 165, 165, 0.68);
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.footer-btn {
  border: 1px solid black;
  border-radius: 20px;
  font-size: 16px;
  padding: 5px 10px;
  transition: all 0.4s ease-in-out;
}

.footer-btn:hover{
  background: black;
  color: white;
  cursor: pointer;
  transition: all 0.4s ease-in-out;
}