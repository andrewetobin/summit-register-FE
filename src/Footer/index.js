import React, { Component } from "react";
import SignRegister from "../SignRegister";
import PropTypes from "prop-types";
import MyMountains from "../MyMountains";
import "./Footer.css";

class Footer extends Component {
  render() {
    const {
      handleSignLog,
      handleSignIn,
      currentDisplay,
      handleMyMountains,
      withinRange,
      handleSignOut,
      isSignedIn
    } = this.props;

    return (
      <footer className="Footer">
        {currentDisplay[0] !== "registerForm" &&
          currentDisplay[0] !== "myMountains" &&
          currentDisplay[0] !== "start" && (
            <SignRegister
              handleSignLog={handleSignLog}
              withinRange={withinRange}
            />
          )}
        {isSignedIn === true && currentDisplay[0] !== "myMountains" && (
          <button className="footer-btn" onClick={handleMyMountains}>
            My Mountains
          </button>
        )}
        {isSignedIn === true && (
          <button className="footer-btn" onClick={handleSignOut}>
            Sign Out
          </button>
        )}
        {this.props.isSignedIn === false && (
          <button className="footer-btn" onClick={handleSignIn}>
            Sign In
          </button>
        )}
      </footer>
    );
  }
}

Footer.propTypes = {};

export default Footer;
