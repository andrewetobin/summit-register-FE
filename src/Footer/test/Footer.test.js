import React from "react";
import ReactDOM from "react-dom";
import Footer from "..";
import { shallow } from "enzyme";

describe("Footer", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Footer
        currentDisplay={["registerForm"]}
        handleSignIn={jest.fn()}
        handleLogUpdate={jest.fn()}
        handleSignLog={jest.fn()}
      />
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const footer = document.createElement("footer");
    ReactDOM.render(
      <Footer
        currentDisplay={["registerForm"]}
        handleSignIn={jest.fn()}
        handleLogUpdate={jest.fn()}
        handleSignLog={jest.fn()}
      />,
      footer
    );
    ReactDOM.unmountComponentAtNode(footer);
  });

  describe("sign register button", () => {
    it("should not display sign register button on sign register screen", () => {
      expect(wrapper.find("SignRegister")).toHaveLength(0);
    });

    it("should display sign register button if not on sign register screen", () => {
      wrapper = shallow(
        <Footer
          currentDisplay={["log"]}
          handleSignIn={jest.fn()}
          handleLogUpdate={jest.fn()}
          handleSignLog={jest.fn()}
        />
      );
      expect(wrapper.find("SignRegister")).toHaveLength(1);
    });
  });
});
