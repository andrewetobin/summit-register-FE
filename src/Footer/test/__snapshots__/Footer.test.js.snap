// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    currentDisplay={
      Array [
        "registerForm",
      ]
    }
    handleLogUpdate={[MockFunction]}
    handleSignIn={[MockFunction]}
    handleSignLog={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <button
          className="sign-in-btn"
          onClick={[MockFunction]}
        >
          SignIn/SignUp
        </button>,
      ],
      "className": "Footer",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SignIn/SignUp",
          "className": "sign-in-btn",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "SignIn/SignUp",
        "type": "button",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <button
            className="sign-in-btn"
            onClick={[MockFunction]}
          >
            SignIn/SignUp
          </button>,
        ],
        "className": "Footer",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SignIn/SignUp",
            "className": "sign-in-btn",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "SignIn/SignUp",
          "type": "button",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
